src/main/java/org/testng/IFactoryMethod.java

```diff
@@ -0,0 +1,41 @@
+package org.testng;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+
+/**
+ * Represents a Factory method that was involved for producing test class instances.
+ */
+public interface IFactoryMethod {
+
+    /**
+     * @return - The class to which the factory method belongs to.
+     */
+    Class<?> getDeclaringClass();
+
+    /**
+     * @return - The name of the factory method. Could either be the constructor's name or the static factory
+     * method's name.
+     */
+    String getName();
+
+    /**
+     * @return - An array of parameters that are part of the factory method.
+     */
+    Class[] getParameterTypes();
+
+    /**
+     * @return - The <code>@Factory</code> annotated method.
+     */
+    Method getMethod();
+
+    /**
+     * @return - The <code>@Factory</code> annotated constructor.
+     */
+    Constructor getConstructor();
+
+    /**
+     * @return - <code>true</code> if the factory method is enabled.
+     */
+    boolean getEnabled();
```

Comment:
`isEnabled`

Reply:
If I change that, then I would have to change the method signature in `ConstructorOrMethod`. We have already exposed this class via `org.testng.ITestNGMethod#getConstructorOrMethod`. I guess that will cause a breakage to users no ?

Reply:
True, it's wrong in `ConstructorOrMethod` but we can deprecate the current method and add a new one.